include/engine/guidance/annotators/textual_route_annotation.hpp

```diff
@@ -0,0 +1,133 @@
+#pragma once
+
+#include "engine/segment_information.hpp"
+#include "engine/guidance/segment_list.hpp"
+#include "extractor/turn_instructions.hpp"
+#include "osrm/json_container.hpp"
+#include "util/bearing.hpp"
+#include "util/cast.hpp"
+
+#include <cstdint>
+
+#include <limits>
+#include <string>
+#include <vector>
+
+namespace osrm
+{
+namespace engine
+{
+namespace guidance
+{
+namespace annotators
+{
+template< typename DataFacadeT >
+inline osrm::json::Array
+AnnotateRoute(const std::vector<SegmentInformation> &route_segments, DataFacadeT* facade)
+{
+    osrm::json::Array json_instruction_array;
+    if( route_segments.empty() )
+      return json_instruction_array;
+    // Segment information has following format:
+    //["instruction id","streetname",length,position,time,"length","earth_direction",azimuth]
+    uint32_t necessary_segments_running_index = 0;
+
+    struct RoundAbout
+    {
+        int32_t start_index;
+        uint32_t name_id;
+        int32_t leave_at_exit;
+    } round_about;
+
+    round_about = {std::numeric_limits<int32_t>::max(), 0, 0};
+    std::string temp_dist, temp_length, temp_duration, temp_bearing, temp_instruction;
+
+    //Generate annotations for every segment
+    for (const SegmentInformation &segment : route_segments)
+    {
+        osrm::json::Array json_instruction_row;
+        TurnInstruction current_instruction = segment.turn_instruction;
+        if (TurnInstructionsClass::TurnIsNecessary(current_instruction))
+        {
+            if (TurnInstruction::EnterRoundAbout == current_instruction)
+            {
+                round_about.name_id = segment.name_id;
+                round_about.start_index = necessary_segments_running_index;
+            }
+            else
+            {
+                std::string current_turn_instruction;
+                if (TurnInstruction::LeaveRoundAbout == current_instruction)
+                {
+                    temp_instruction =
+                        std::to_string(cast::enum_to_underlying(TurnInstruction::EnterRoundAbout));
+                    current_turn_instruction += temp_instruction;
+                    current_turn_instruction += "-";
```

Comment:
Do we want to force this representation here in this function? Maybe make it a policy with a default value that makes sense.